lib/fog/vsphere/requests/compute/request.rb

```diff
@@ -0,0 +1,166 @@
+require 'fog/vsphere/requests/request'
+
+module Fog
+  module Vsphere
+    module Requests
+      module Compute
+        class Request < Fog::Vsphere::Requests::Request
```

Comment:
This file is the worst place of whole thing.
{Not mentioning the naming as I expect discussion about that.}
It creates a dependency cointainer, but I believe we could narrow it down by the commands calling one another, once we encapsulate them all, as that is what whole file basically does now anyways.
And I kind of like the idea as it would be more explicit dependency.
Unlike now, when we call other requests private methods.